app/models/container_template.rb

```diff
@@ -34,11 +34,23 @@ def instantiate(params, project = nil)
                                             :namespace => project
                                           },
                                           :objects    => objects,
-                                          :parameters => params)
+                                          :parameters => params_to_hashes(params))
     create_objects(processed_template['objects'], project)
     @created_objects.each { |obj| obj[:miq_class] = MIQ_ENTITY_MAPPING[obj[:kind]] }
   end
 
+  def params_to_hashes(params)
+    params.collect do |param|
+      {
+        :name     => param.name,
+        :value    => param.value,
+        :generate => param.generate,
+        :from     => param.from,
+        :required => param.required
+      }
+    end
+  end
+
```

Comment:
This is a helper method, and should be added to the bottom of the file under a `private` so it doesn't get used publicly. 

There's some duplication here, you could slice attributes to maintain a single list of the ones you want doing something like...

```ruby

def params_to_hashes(params)
  keys = %w(name value generate from required)

  params.each_with_object([]) do |param, collection|
    collection << param.attributes.slice(*keys) 
  end
end
```